src/qt/bitcoingui.cpp

```diff
@@ -946,12 +945,25 @@ void BitcoinGUI::changeEvent(QEvent *e)
             QWindowStateChangeEvent *wsevt = static_cast<QWindowStateChangeEvent*>(e);
             if(!(wsevt->oldState() & Qt::WindowMinimized) && isMinimized())
             {
+                if (wsevt->oldState() & Qt::WindowMaximized)
+                {
+                    QTimer::singleShot(0, this, SLOT(showMaximized()));
+                }
+                else if (wsevt->oldState() & Qt::WindowFullScreen)
+                {
+                    QTimer::singleShot(0, this, SLOT(showFullScreen()));
+                }
+                else
+                {
+                    QTimer::singleShot(0, this, SLOT(showNormal()));
+                }
                 QTimer::singleShot(0, this, SLOT(hide()));
                 e->ignore();
             }
         }
     }
 #endif
+    QMainWindow::changeEvent(e);
```

Comment:
Why?

Reply:
I thought, it makes senses to reorder. But I see, it makes no sense. Fixed.

Reply:
You could have some reason, otherwise it's unrelated to your goal.

Reply:
The idea was, that theoretically we don't know what QMainWindow::changeEvent does. So if I want to add something "on top" of the default behaviour, I have to do it after that and not before. But this is kind of a academical thought. To be honest, I am not a Qt expert.